sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAsyncApiTests.java

```diff
@@ -424,12 +433,3698 @@ public void createOptionsWithTimeToExpireRelativeToNow() {
+        createdFiles.add(testFile);
+
+        if (!testFile.exists()) {
+            assertTrue(testFile.createNewFile());
+        }
+
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+
+        Set<OpenOption> openOptions = new HashSet<>(Arrays.asList(
+            StandardOpenOption.CREATE, StandardOpenOption.READ, StandardOpenOption.WRITE));
+
+        StepVerifier.create(fc.readToFileWithResponse(testFile.getPath(), null, null,
+                null, null, false, openOptions))
+                .assertNext(r -> {
+                    try {
+                        assertEquals(DATA.getDefaultText(), new String(Files.readAllBytes(testFile.toPath()), StandardCharsets.UTF_8));
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                    .verifyComplete();
+    }
+
+    @Test
+    public void downloadFileExistOpenOptions() throws IOException {
+        File testFile = new File(prefix + ".txt");
+        testFile.deleteOnExit();
+        createdFiles.add(testFile);
+
+        if (!testFile.exists()) {
+            assertTrue(testFile.createNewFile());
+        }
+
+        fc.append(DATA.getDefaultBinaryData(), 0).block();
+        fc.flush(DATA.getDefaultDataSizeLong(), true).block();
+
+        Set<OpenOption> openOptions = new HashSet<>(Arrays.asList(StandardOpenOption.CREATE,
+            StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.READ, StandardOpenOption.WRITE));
+
+        StepVerifier.create(fc.readToFileWithResponse(testFile.getPath(), null, null,
+                null, null, false, openOptions))
+            .assertNext(r -> {
+                try {
+                    assertEquals(DATA.getDefaultText(), new String(Files.readAllBytes(testFile.toPath()), StandardCharsets.UTF_8));
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            })
+            .verifyComplete();
+    }
+
+    @EnabledIf("com.azure.storage.file.datalake.DataLakeTestBase#isLiveMode")
+    @ParameterizedTest
+    @MethodSource("downloadFileSupplier")
+    public void downloadFile(int fileSize) {
+        File file = getRandomFile(fileSize);
+        file.deleteOnExit();
+        createdFiles.add(file);
+
+        fc.uploadFromFile(file.toPath().toString(), true).block();
+        File outFile = new File(testResourceNamer.randomName("", 60) + ".txt");
+        outFile.deleteOnExit();
+        createdFiles.add(outFile);
+
+        if (outFile.exists()) {
+            assertTrue(outFile.delete());
+        }
+
+        StepVerifier.create(fc.readToFileWithResponse(outFile.toPath().toString(), null,
+                new ParallelTransferOptions().setBlockSizeLong(4L * 1024 * 1024),
+         null, null, false, null))
+            .assertNext(r -> {
+                compareFiles(file, outFile, 0, fileSize);
```

Comment:
I'd move the `compareFiles` call to outside the StepVerifier call, anything interacting with IO needs extra care when dealing with Reactor to prevent blocking threads.